pkg/sdk/validation/manifest/json_remote_implementation.go

```diff
@@ -93,10 +115,106 @@ func (v *RemoteImplementationValidator) Do(ctx context.Context, _ types.Manifest
 			})
 		}
 	}
-
 	return checkManifestRevisionsExist(ctx, v.hub, manifestRefsToCheck)
 }
 
+func (v *RemoteImplementationValidator) validateInputArtifactsNames(ctx context.Context, entity types.Implementation) (ValidationResult, error) {
+	var validationErrs []error
+	var interfacesInputNames []string
+	var implAdditionalInput []string
+	var workflowArtifacts []wfv1.Artifact
+
+	//1. get interface input names
+	for _, implementsItem := range entity.Spec.Implements {
+		interfaceInput, err := v.fetchInterfaceInput(ctx, hubpublicgraphql.InterfaceReference{
+			Path:     implementsItem.Path,
+			Revision: implementsItem.Revision,
+		}, v.hub)
+		if err != nil {
+			return ValidationResult{}, errors.Wrap(err, "while fetching Interface inputs")
+		}
+		for _, inputParameter := range interfaceInput.Parameters {
+			interfacesInputNames = append(interfacesInputNames, inputParameter.Name)
+		}
+		for _, inputTypeInstance := range interfaceInput.TypeInstances {
+			interfacesInputNames = append(interfacesInputNames, inputTypeInstance.Name)
+		}
+	}
+
+	//2. get implementation additional inputs
+	if entity.Spec.AdditionalInput != nil {
+		for name := range entity.Spec.AdditionalInput.Parameters {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+		for name := range entity.Spec.AdditionalInput.TypeInstances {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+	}
+
+	//3. get inputs from entrypoint workflow template
+	workflow, err := decodeImplArgsToArgoWorkflow(entity.Spec.Action.Args)
+	if err != nil {
+		return ValidationResult{}, errors.Wrap(err, "while decoding Implementation arguments to Argo workflow")
+	}
+	if workflow != nil && workflow.WorkflowSpec != nil {
+		idx, err := argo.GetEntrypointWorkflowIndex(workflow)
+		if err != nil {
+			return ValidationResult{}, errors.Wrap(err, "while getting entrypoint index from workflow")
+		}
+		workflowArtifacts = append(workflowArtifacts, workflow.Templates[idx].Inputs.Artifacts...)
```

Comment:
Why do you copy the input artifacts? We just iterate over them, so can't we do simply

```suggestion
		workflowArtifacts = workflow.Templates[idx].Inputs.Artifacts
```

?

Reply:
good point :+1: it was leftover from the previous approach